formbricks/formbricks
# feature/auto close survey

## What does this PR do?



Adds the option to set an autoClose delay in seconds, that closes the in-product survey if it is not hovered or clicked within the specified time. Also prepared the logic to include in the survey preview.


![image](https://github.com/formbricks/formbricks/assets/42251569/df7591ba-95af-4e93-a89d-f28b76127430)

![image](https://github.com/formbricks/formbricks/assets/42251569/2bbfd928-bc72-4169-94da-146c485879de)


## Type of change




## How should this be tested?



- Check if it isnt closing if autoClose is off
- Check if it is closing after x seconds for a few times
- Check if it isnt closing if the survey is hovered within the time 

## Checklist



- [x] I have read the [contributing guide](https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md)
- [x] I have added comments to my code, particularly in hard-to-understand bits
- [x] I ran `pnpm build` and `pnpm lint` and checked for build and linting errors
- [x] My contribution does not cause any warnings
- [x] I have removed all `console.logs`
- [x] I have merged the latest changes from main onto my branch with `git pull origin main`
- [x] My contribution causes changes to the Formbricks Docs
